docs/code-i18n.md

```diff
@@ -0,0 +1,31 @@
+# Code Internationalization
+
+When a project is intended to be localized to multiple languages, all the strings in the codebase must also be translated. Any language strings that are "hardcoded" in PHP or JavaScript files must follow the code internationalization practices to ensure they can be translated into any language you want to support.
+
+This is done by passing all strings that your custom code outputs through the translation API functions. This is used to both swap your strings for their translated variant at runtime depending on the language of the site, and also be used as markers to generating translation template files (POT files).
+
+Once you have generated a POT file for you custom code, you can provide your translators with these files. Typically programs such as [poedit](https://poedit.net/) are used to author translations.
+
+## Translation API Functions
+
+Whenever you output a string in code, you must use a translation function. HM Platform uses the [WordPress Translation API](https://codex.wordpress.org/I18n_for_WordPress_Developers) to provide code level internationalization (i18n). Translations in code are tied to a "Text Domain" which is an identifier for which translation file should be used for a given translation. It's recommended you use a single translation domain for all your project's custom code. This will mean you only need to get a single POT file translated.
+
+See [Translatable Strings](https://codex.wordpress.org/I18n_for_WordPress_Developers#Translatable_strings) for details on the functions you should use, depending on context.
```

Comment:
Think it's worth adding some recommendations here eg.

- text domain must always be a string, never a constant or variable
- favour translation functions that provide context for strings such as `_x()` to help translators provide the best translation
- always use the `esc_html` prefixed version when outputting text directly

Given the translation functions are pretty stable maybe we should add some code examples (especially for `__()` vs `esc_html__()` or document them directly?